### PR TITLE
Fix Nefarian Gate Missing #12434

### LINKED ISSUE #12434: [BWL]Nefarian Gate missing
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

There should be a gate that is closing on the fight start against nefarian that is missing.

### Expected Blizzlike Behaviour

gate existing and closing during boss fight

### Source

https://youtu.be/IhkMiNfYmyE?t=939

### Steps to reproduce the problem

pull nefarian
see gate missing

### Extra Notes

originally reported: https://github.com/chromiecraft/chromiecraft/issues/3719

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/0ef7f999db7284c2a7776b70c9ecf008d8551f16

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by SoglaHash**
## Changes Proposed:
-  [BWL]: Add missing Nefarian's gate

## Issues Addressed:

- Closes #12434
- Closes https://github.com/chromiecraft/chromiecraft/issues/3719

- Gate gameobject (id: 176966) was missing. The gate is open by default
and will close shortly after Nefarian's dialogue is complete. It will
open once the fight resets. The game is also not interactible by
players.

- I renamed the "PORTCULLIS_NEFARIAN" (gate that opens after killing
Chrom) to "PORTCULLIS_CHROMAGGUS_EXIT" as
the name was confusing.

## SOURCE:

what it should look like: https://youtu.be/IhkMiNfYmyE?t=939

## Tests Performed:

Built with the docker setup without issue

https://user-images.githubusercontent.com/74299960/182498382-011a9fdc-b559-4d29-bac0-ef31115eff33.mp4

## How to Test the Changes:

1. .go c 85785
2. enter/reset combat and observe gate

## Known Issues and TODO List:

- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
Would also close https://github.com/chromiecraft/chromiecraft/issues/3719.

**Comment by balleny** [comment]
does this Gate Include Line of Sight, preventing the mobs spawned during the fight to not pass through it (including the boss nefarian it self)?
if so (which would be correct), it should also abort the boss fight, if no target can be attacked. Otherwise, P1 of the fight would keep running and spawn mobs without limit.
Also, gate needs to open in case of a wipe.

**Comment by SoglaHash** [comment]
> does this Gate Include Line of Sight, preventing the mobs spawned during the fight from passing through it (including the boss nefarian it self)? if so (which would be correct), it should also abort the boss fight, if no target can be attacked within the enclosed area. Otherwise, P1 of the fight would keep running and spawn mobs without limit. Also, gate needs to open in case of a wipe.

This gate includes line of sight for player preventing them from aiding their allies or hitting mobs.  
Nefarian will cast through and mobs will walk through. Players right outside the gate, Chrom room or Flamegor/Ebonroc room will enter combat and be chased.   

After wiping, the Chrom gate (right on picture) and Nefarian gate re-open. I believe closing the Chrom gate room together with Nef gate is blizzlike and was already implemented, so I did not change it.

**Comment by Maelthyrr** [comment]
Tested and work fine.

- [X] Door close when the fight is engaged
- [X] Door reopen is the fight is reset

**Review by Nefertumm — APPROVED** [review]

**Comment by Nefertumm** [comment]
Thank you so much for your first contribution here! 🥳

File: data/sql/updates/pending_db_world/rev_1659484347526582712.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+DELETE FROM `gameobject` WHERE `id` = 176966;
+INSERT INTO `gameobject` (`id`, `map`, `zoneId`, `areaId`,
+    `spawnMask`, `phaseMask`, `position_x`, `position_y`, `position_z`,
+    `orientation`, `rotation0`, `rotation1`, `rotation2`, `rotation3`,
+    `spawntimesecs`, `animprogress`, `state`, `ScriptName`, `VerifiedBuild`)
+VALUES (176966, 469, 0, 0, 1, 1, -7488.1, -1150.7, 476.535, 3.73064, -0.0,
+    -0.0, -0.95694, 0.290285, 300, 0, 1, 'Nefarian\'s Gate', 0);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1659484347526582712.sql:7]: Oh I just noticed you made a mistake here, putting as scriptname 'Nefarian\'s Gate'. I'm too late now :/

**Pull request merged**